plugin/internal/command.go

```diff
@@ -0,0 +1,148 @@
+package internal
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/puppetlabs/wash/activity"
+)
+
+// Command is a wrapper to exec.Cmd. It handles context-cancellation cleanup
+// and defines a String() method to make logging easier.
+type Command struct {
+	*exec.Cmd
+	ctx           context.Context
+	pgid          int
+	waitResult    error
+	waitDoneCh    chan struct{}
+	waitOnce      sync.Once
+}
+
+// NewCommand creates a new command object that's tied to the passed-in
+// context. When cmd.Start() is invoked, the command will run in its
+// own process group. When the context is cancelled, a SIGTERM signal will
+// be sent to the command's process group. If after five seconds the command's
+// process has not been terminated, then a SIGKILL signal is sent to the
+// command's process group.
+func NewCommand(ctx context.Context, cmd string, args ...string) *Command {
+	if ctx == nil {
+		panic("plugin.newCommand called with a nil context")
+	}
+	cmdObj := &Command{
+		Cmd:           exec.Command(cmd, args...),
+		ctx:           ctx,
+		pgid:          -1,
+		waitDoneCh:    make(chan struct{}),
+	}
+	cmdObj.SysProcAttr = &syscall.SysProcAttr{
```

Comment:
The PGID code was ripped off https://medium.com/@felixge/killing-a-child-process-and-all-of-its-children-in-go-54079af94773